dss/api/files.py

```diff
@@ -121,12 +120,22 @@ def get_helper(uuid: str, replica: Replica, version: str = None, token: str = No
     return response
 
 
-def _verify_checkout(
-        replica: Replica, token: typing.Optional[str], file_metadata: dict, blob_path: str,
-) -> typing.Tuple[str, bool]:
+def _verify_checkout(replica: Replica, file_metadata: dict, blob_path: str,) -> typing.Tuple[str, bool]:
     cloud_handle = Config.get_blobstore_handle(replica)
     hca_handle = Config.get_hcablobstore_handle(replica)
 
+    # Lookup existing checkout job. token is an exception if checkout failed
+    async_key = f"{replica.name}/{blob_path}"
+    token = AsyncStateItem.get(async_key)
+    if isinstance(token, S3CopyEtagError):
+        raise DSSException(
+            requests.codes.unprocessable,
+            "missing_checksum",
+            "Incorrect s3-etag"
+        )
+    elif isinstance(token, AsyncStateError):
+        raise token
```

Comment:
Once a checkout has failed (e.g. due to a checksum error), it is never possible for a user to get the file again without administrative intervention to clear the error from the DynamoDB table.
We had some good discussion regarding desired behavior options and some additional/alternate ways of handling this.